### DISCUSSION TITLE
Cannot prototype for a migration if you are going to customize it

### DISCUSSION THREAD

**Discussion by hardysabs2** [prisma/prisma#21516] [Q&A]
### Question

If you want to customize a migration step ONE here
https://www.prisma.io/docs/guides/migrate/developing-with-prisma-migrate/customizing-migrations
MUST be done in one, it cannot be done using prototyping 
https://www.prisma.io/docs/concepts/components/prisma-migrate/db-push
because after one or more
prisma db push
a call to npx prisma migrate dev --create-only
detects drift and says continuing will lose all data
If you select Yes
It clears data and rebuilds the schema
But the seed fails due to the schema change
So we have an unapplied migration which we then carefully adapt to accommodate production data
But we are unable to test it  because our database has no data.

This was my workaround:
edit schema
prisma db push
try it
edit schema
prisma db push
try it - happy
cp ./prisma/schema.prisma ./prisma/schema.prisma.new
revert ./prisma/schema.prisma
npx prisma migrate reset
cp ./prisma/schema.prisma.new ./prisma/schema.prisma
npx prisma migrate dev --create-only
edit to accommodate prod data safely
prisma migrate deploy (to test)

This feels very cluncky - is there a better solution?

e.g. npx prisma migrate  --create-only (just create the migration from the schema change)

Better still would be to run seed at a given point and then all remaining migrations are implemented production safe - not reliant on seeding

Thanks,

Andrew

**Comment by ludralph** [comment]
Hi @hardysabs2 👋 

Are you trying to maintain production data that cannot be reset? If yes, then using [Baselining](https://www.prisma.io/docs/orm/prisma-migrate/workflows/baselining) would be the best possible approach to take instead of the workaround you mentioned.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
